### PR TITLE
adding a bit of error handling, cleaning up code, and updating docs

### PR DESCRIPTION
**Pull request by cantupaz**
Hi. I added a little bit of error handling when arming/disarming. I got rid of the ENABLE_CODE flag that was probably confusing users in the config window (I found it confusing that I could leave the code empty and I also had to uncheck the box to get rid of the PIN). 

I added a few missing type hints and updated the readme file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,13 +14,27 @@ securitas:
...
-- Added a new configuration panel to change some of the options that you set during the setup. Here is an screenshot.
+- Lists all your installations and add a panel into Home Assistant.
+- Supports Sentinel and adds temperature and humidity sensors for each Sentinel in each installation you have.
+- Supports installations with Perimetral (external) alarms.
+
+## Setup
+![Options](./docs/images/setup.png)
+
+- Enter the username and password for your Securitas account.
+- Use 2FA (default: no). It is not necessary to use two-factor authentication to use this integration, but we're leaving that option in case Securitas changes that in the future.
```
Comment (guerrerotook) [inline README.md:25]: Hey @cantupaz I think that the default should be yes, enabled by default. I think all countries are actually enabling 2FA right now. That how it was in the past, and the API stopped working. I don't see Securitas not enabling the 2FA or even disabling this. I don't know where you are getting this information from, but this should be enabled by default.

File: custom_components/securitas/__init__.py [hunk 2]
```diff
@@ -50,41 +50,19 @@
 CONF_DEVICE_INDIGITALL = "idDeviceIndigitall"
 CONF_ENTRY_ID = "entry_id"
 CONF_INSTALLATION_KEY = "instalation"
-CONF_ENABLE_CODE = "enable_code"
 CONF_DELAY_CHECK_OPERATION = "delay_check_operation"
 
 DEFAULT_USE_2FA = False
```
Comment (guerrerotook) [inline custom_components/securitas/__init__.py:55]: This should be true, look at my previous comment.

**Review by guerrerotook — CHANGES_REQUESTED** [review]

**Comment by cantupaz** [comment]
I can try to get to this tomorrow, but at least in Spain the 2FA doesn’t do anything. Of course, in the official app, it won’t let you proceed without 2FA, but the API returns the authorization token just after login. You get the exact same token after the 2FA.

**Comment by guerrerotook** [comment]
That is not true @cantupaz, there are a lot of issues people complaining that it logged but then after some time the integration stop working, https://github.com/guerrerotook/securitas-direct-new-api/issues/52 here in Spain, France and other countries as well. 

Again, I don't know exactly where you got that it doesn't do anything. My recommendation, for everyone should be to enable this and as you said, it this fails for you have the option to disable, but not the other way around. This works for you without the 2FA fine, but we have had a lot of issues in the past with this and I believe that not enabling this will likely result in more people opening issues because weird errors that all down to not be authenticated.

**Comment by guerrerotook** [comment]
Just to be clear, I only want you to change the default value to true, that's it. The rest of the PR is great!

**Comment by cantupaz** [comment]
Easy. I said I’ll change it, but probably tomorrow because I got something to do today. Btw, I changed that months ago 😊 I agree it’s safer to switch the default in case one day Securitas stops sending the token after login.

**Comment by guerrerotook** [comment]
Thanks very much!, don't worry about when, take your time.

**Comment by cantupaz** [comment]
I made the changes, please take a look.

**Review by guerrerotook — APPROVED** [review]
Thanks very much!

**Pull request merged**